spec/rails_helper.rb

```diff
@@ -70,28 +70,37 @@
   config.include S3Utils
 
   config.before(:suite) do
-    Dir.glob('{drivers,spec}/**/fixpoints/*.yml').each do |filename|
-      FileUtils.rm(filename)
+    # disable paper trail for test performance
+    PaperTrail.enabled = false
+
+    # disable rack attack unless we are testing it explicitly
+    Rack::Attack.enabled = false
+
+    example_file_paths = RSpec.world.filtered_examples.values.flatten.map { |e| e.metadata[:file_path] }.uniq
+    # load fixpoints if we're running tests that may need them
+    if example_file_paths.grep(%r{/drivers/(hud_path_report|hud_spm_report|hud_data_quality_report)/}).any? # rubocop:disable Style/RegexpLiteral
+      Dir.glob('{drivers,spec}/**/fixpoints/*.yml').each do |filename|
+        FileUtils.rm(filename)
+      end
+      Dir.glob('{drivers,spec}/**/fixpoints/*.sql').each do |filename|
+        # The 2024 test kit is HUGE, so we'll use the preprocessed version
+        # This means when the test kit changes, we need to update both the CSVs and the SQL fixpoint
+        next if filename.include?('drivers/datalab_testkit/spec/fixpoints')
+
+        FileUtils.rm(filename)
+      end
+      GrdaWarehouse::WarehouseReports::ReportDefinition.maintain_report_definitions
     end
-    Dir.glob('{drivers,spec}/**/fixpoints/*.sql').each do |filename|
-      # The 2024 test kit is HUGE, so we'll use the preprocessed version
-      # This means when the test kit changes, we need to update both the CSVs and the SQL fixpoint
-      next if filename.include?('drivers/datalab_testkit/spec/fixpoints')
 
-      FileUtils.rm(filename)
+    # load hmis forms if we're testing the driver
+    if example_file_paths.grep(%r{/drivers/hmis/}).any? # rubocop:disable Style/RegexpLiteral
+      ::HmisUtil::JsonForms.seed_all if ENV['ENABLE_HMIS_API'] == 'true'
     end
 
-    GrdaWarehouse::Utility.clear!
-    Delayed::Job.delete_all
-    GrdaWarehouse::WarehouseReports::ReportDefinition.maintain_report_definitions
+    # I'm not sure that we need these on every test run
+    # GrdaWarehouse::Utility.clear!
+    # Delayed::Job.delete_all
```

Comment:
If the tests are passing without them, then we don't.  There were some tests, historically, that didn't cleanup well, so this was intended to catch that.  But if we've reached a stables state, it would be great to remove them.